_posts/2024-01-05-devcontainer-platform-agnostic-team.md

```diff
@@ -0,0 +1,68 @@
+
+From the perspective of a team and company, especially one that works on open source government software, being able to support engineers who use Windows, Linux and Mac allows the company to recruit people from a wider variety of industries and experiences. I didn’t have to, for example, retrain and learn the Windows environment to work on government apps, which often are built by Windows-only teams.
+
+As an engineer who focuses primarily on the front-end user experience, I feel that our team is stronger because we are testing the app daily on Windows, Linux and Mac, across various browsers. I no longer have to click through an emulator took like BrowserStack to check how the app looks and feels on Windows. Windows users aren’t second-class, and vice versa. A common criticism of all-Apple engineering and design teams is that they are not working on the hardware that most Americans can afford and use, and that’s not a problem here at Compiler.
+
+Often times, our team supports front-line government IT teams over Zoom, and having a team that understands all the major computing platforms is also essential for us to help debug issues on clients’ machines.
+
+## Technical challenges and looking forward
+
+So what are the technical challenges? I’ll be the first to admit that setting up and configuring a Dockerized container may be very challenging for a junior engineer - depending on how complex the application is, that might require a more senior engineer for the initial set up.
+
+If you’re the kind of developer who has a very specific code editor - Neovim, Emacs or what have you - and a very custom suite of extensions and plugins, this way of working may not be for you. Having the command line and code editor integrated together might also not be everyone’s cup of tea.
+
+One of the greatest challenges our team has faced is running Cypress tests. The Benefits application has a suite of end-to-end Cypress tests that run a browser and click through the app – but we have yet to figure out how to get that all running in the container. We have to run Cypress outside of the container, which means downloading Node, npm, Cypress and its dependencies on our local machines. We’re continuing to investigate how to best achieve a fully-containerized end-to-end front-end browser testing suite. (If you have any experience or thoughts on this, feel free to chime in on our [corresponding GitHub issue](https://github.com/cal-itp/benefits/issues/681).) Perhaps a post in this coming year elucidating this configuration challenge is in the future.
```

Comment:
The last sentence reads a little funny. One suggestion:

```diff
- Perhaps a post in this coming year elucidating this configuration challenge is in the future.
+ Perhaps a post elucidating this configuration challenge will be forthcoming this year.
```

Reply:
Rewrote the last part: https://github.com/compilerla/compiler.la/pull/156/commits/45cb3de4b73584e1f1d198fcdfe06e42768c5858